### PR TITLE
added roles necessary to put docker and nextcloud on debian machine in 2169 Mission known as -cloud-

### PR DESCRIPTION
**Pull request by danhimalplanet**
install docker-ce on cloud

write docker-compose necessary to run nextcloud image

it doesnt automatically start nextcloud, for now.

### CODE REVIEW

File: roles/docker-host/handlers/main.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+- name: apt-update
```
Comment (SuperQ) [inline roles/docker-host/handlers/main.yml]: This isn't necessary.

File: roles/docker-host/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+---
+- name: Install packages
```
Comment (SuperQ) [inline roles/docker-host/tasks/main.yml]: I think this is missing the apt repo setup.

```yaml
- name: Add Docker's GPG key
  apt_key:
    id: "9DC858229FC7DD38854AE2D88D81803C0EBFCD88"
    url: "https://download.docker.com/linux/ubuntu/gpg"

- name: Configure docker APT repository
  apt_repository:
    repo: "deb [arch=amd64] https://download.docker.com/linux/{{ ansible_distribution | lower }} {{ ansible_distribution_release }} stable"
docker_apt_cache_time: 86400
```

**Comment by SuperQ** [comment]
One idea, for docker setup, we might consider importing an existing docker-ce setup role.

Maybe https://github.com/nickjj/ansible-docker

File: roles/docker-host/handlers/main.yml (hunk 1 above)
Comment (danhimalplanet) [reply roles/docker-host/handlers/main.yml]: removed

**Comment by danhimalplanet** [comment]
> One idea, for docker setup, we might consider importing an existing docker-ce setup role.
> 
> Maybe https://github.com/nickjj/ansible-docker

added to submodules
i made role nextcloud-docker require nickjj.docker

File: site.yml [hunk 3]
```diff
@@ -70,3 +70,12 @@
...
     - postfix
     - mailman3
+
+- name: Nextcloud on debian box in space
+  hosts: cloud
+  become: yes
+  become_method: sudo
+  tags: [cloud]
+  roles:
+    - utils
```
Comment (SuperQ) [inline site.yml]: You don't need utils, this is in the `common` tag applied to all nodes.

Comment (danhimalplanet) [reply site.yml]: fixed

File: .gitmodules [hunk 4]
```diff
@@ -13,3 +13,9 @@
 [submodule "roles/ansiblebit.grafana"]
 	path = roles/ansiblebit.grafana
 	url = https://github.com/ansiblebit/grafana.git
+[submodule "roles/nickjj.docker"]
+	path = roles/nickjj.docker
+	url = https://github.com/nickjj/ansible-docker
+[submodule "ansible-docker"]
```
[the remark this answers is not in the document]
Comment (danhimalplanet) [reply .gitmodules]: fixed

**Comment by SuperQ** [comment]
So this is maybe fine for now, but we should probably convert the Makefile stuff to standard Ansible [docker_service](https://docs.ansible.com/ansible/2.6/modules/docker_service_module.html).

**Pull request merged**